docs/reference/modules/cross-cluster-search.asciidoc

```diff
@@ -273,3 +279,45 @@ GET /cluster_one:twitter,cluster_two:twitter,twitter/_search <1>
 // TESTRESPONSE[s/"_score": 1/"_score": "$body.hits.hits.0._score"/]
 // TESTRESPONSE[s/"_score": 2/"_score": "$body.hits.hits.1._score"/]
 <1> The `clusters` section indicates that one cluster was unavailable and got skipped
+
+[float]
+[[ccs-reduce-mode]]
+=== Reduce mode
+
+When searching across remote clusters, the execution path may differ depending
+on the selected reduce mode:
+
+- `remote`: the coordinating node sends one search request to each cluster.
+Each cluster performs the search independently, reducing and fetching results
+which are then returned to the caller. Once the CCS node has received all the
+responses, it performs another reduction and returns the relevant results back
+to the user. The `remote` reduce mode is beneficial when there is network
+latency between the coordinating node and the remote clusters involved, which
+is typically the case. A single request is sent to each remote cluster, at the
+cost of retrieving `from` + `size` already fetched results.
+
+- `local`: the coordinating node sends a <<search-shards,search shards>>
+request to each remote cluster, in order to collect information about their
+corresponding remote indices involved in the search request and the shards
+where their data is located. Once each cluster has responded to such request,
+the search executes as if all shards were part of the same cluster. The
+coordinating node sends one request to each shard involved, each shard
+executes the query and returns its own results which are then reduced (and
+fetched, depending on the <<search-request-search-type, search type>>) by the
+coordinating node. The `local` reduce mode is beneficial whenever there is very
+ low network latency between the coordinating node and the remote clusters
+ involved, as it treats all shards the same, at the cost of sending many
+ requests to each remote cluster.
+
+- `auto`: `remote` is used whenever possible. In case a scroll is provided,
```

Comment:
so what happens if I specify `remote` and I have inner hits?

Reply:
I wonder if we have to have a `mode` at all. can't we simply go with `local_reduce: true|false` if it's true we force local if not we try remote if possible?

Reply:
> so what happens if I specify remote and I have inner hits?

If you specify remote and you provide a scroll or request inner hits as part of fields collapsing, you get back an error. I think it's better to let users know that what they have requested is not supported when they are explicit in their request. 

> I wonder if we have to have a mode at all. can't we simply go with local_reduce: true|false if it's true we force local if not we try remote if possible?

I think that having remote (or false) act only as a preference is confusing, as users are requesting something that may not be possible. Ideally, if somebody sets `local_reduce: false` (or `reduce_mode: remote`), we would never go and fan out to the shards. It sounds weird to me to have an option act as a proper on/off behaviour and the other one as a preference, unless they have different names.

If we want to have only two values and get rid of the `auto` option, we could have `reduce_mode: always_local | prefer_remote` which I find  much more self-explanatory than a boolean. Yet I am not sure it is much different from what we have now in the PR.

Reply:
or we could also remote `auto` to `prefer_remote`.

Reply:
we discussed this a bit with @jimczi and the team, we could go with a `boolean` but I would like to have some `force` part in the name of the option. We could go with `ccs_force_local_reduce` which defaults to false, or maybe even `ccs_optimize_for_latency` which defaults to true? I think I prefer the latter as it tells when such option should be used. Any other thoughts/preference?